pkg/cloudprovider/providers/azure/azure_storage.go

```diff
@@ -19,53 +19,80 @@ package azure
 import (
 	"fmt"
 
+	"github.com/Azure/azure-sdk-for-go/arm/storage"
+	"github.com/Azure/go-autorest/autorest/to"
 	"github.com/golang/glog"
 )
 
+const (
+	defaultStorageAccountType  = string(storage.StandardLRS)
+	fileShareAccountNamePrefix = "f"
+)
+
 // CreateFileShare creates a file share, using a matching storage account
-func (az *Cloud) CreateFileShare(name, storageAccount, storageType, location string, requestGB int) (string, string, error) {
-	var errResult error
-	accounts := []accountWithLocation{}
-	if len(storageAccount) > 0 {
-		accounts = append(accounts, accountWithLocation{Name: storageAccount})
-	} else {
-		// find a storage account
-		accounts, errResult = az.getStorageAccounts()
-		if errResult != nil {
-			// TODO: create a storage account and container
-			return "", "", errResult
+func (az *Cloud) CreateFileShare(shareName, accountName, accountType, location string, requestGB int) (string, string, error) {
+	if len(accountName) == 0 {
+		// find a storage account that matches accountType
+		accounts, err := az.getStorageAccounts(accountType)
+		if err != nil {
+			return "", "", fmt.Errorf("could not list storage accounts for account type %s: %v", accountType, err)
 		}
-	}
-	for _, account := range accounts {
-		glog.V(4).Infof("account %s type %s location %s", account.Name, account.StorageType, account.Location)
-		if ((storageType == "" || account.StorageType == storageType) && (location == "" || account.Location == location)) || len(storageAccount) > 0 {
-			// find the access key with this account
-			key, innerErr := az.getStorageAccesskey(account.Name)
-			if innerErr != nil {
-				errResult = fmt.Errorf("could not get storage key for storage account %s: %v", account.Name, innerErr)
-				continue
+
+		if len(accounts) > 0 {
+			for _, account := range accounts {
+				glog.V(4).Infof("account %s type %s location %s", account.Name, account.StorageType, account.Location)
+				if (accountType == "" || account.StorageType == accountType) && (location == "" || account.Location == location) {
```

Comment:
The accountType check part could indeed be removed now, since it just runs in getStorageAccounts.

Actually we can also let getStorageAccounts to do 'location' check, this eliminates unneeded account returned from that function. Then we only need to take `accounts`'s first element here if exists.

Reply:
good point, fixed